xarray/backends/rasterio_.py

```diff
@@ -74,7 +74,7 @@ def __getitem__(self, key):
                     raise IndexError(_ERROR_MSG)
             window.append((start, stop))
 
-        out = self.rasterio_ds.read(band_key, window=window)
```

Comment:
will this change work with rasterio < v1.0?

Reply:
The [docstring](https://github.com/mapbox/rasterio/blob/96e63f02b5929b43db56a79e06efd748b8d89fa9/rasterio/_io.pyx#L674) for 0.36 says it takes a tuple of pairs of ints, so it should work (as it did on my machine). I've added another environment file specific to rasterio<=1.0 to test these conditions.

Reply:
Actually the default should be to test against 0.36 (which is the current stable version) and to add an "allowed failure" test for 1.Xa in the matrix. But maybe I'm being pedantic here...

Reply:
I'm +1 on @fmaussion's idea here. I think we want to test against 0.36 until rasterio has an official 1.0 release. We can add an "allowed failure" test to integrate 1.0 into our test suite. 

The good news is that this change seems to work for both versions so there's not anything huge to worry about here.

Reply:
Makes sense and certainly reduced what I needed to change! I added a new entry to the test matrix for py36 and rasterio>=1.* from the conda-forge development channel.

Unfortunately I don't think Travis sees the new matrix entry. Not sure if it's related, but I backed out the commits that had changed many of the environment files, hoping to tidy up all the modifications. Sorry if this complicates the PR.

Reply:
> Unfortunately I don't think Travis sees the new matrix entry

You need to add the entry to both the `matrix` and `allow_failures` lists

Reply:
Thanks for catching this! Should be fixed now.